src/ray/raylet/node_manager.cc

```diff
@@ -1252,10 +1247,10 @@ void NodeManager::TreatTaskAsFailed(const Task &task, const ErrorType &error_typ
           current_time_ms()));
     }
   }
-  // A task failing is equivalent to assigning and finishing the task, so clean
-  // up any leftover state as for any task dispatched and removed from the
-  // local queue.
-  lineage_cache_.AddReadyTask(task);
```

Comment:
Is it the case that in order for a task to reach this point, we must have called `SubmitTask` on it?